CardsCashCasino/Manager/TexasHoldEmManager.cs

```diff
@@ -810,199 +813,249 @@ private static void OnTimeoutEvent(object source, ElapsedEventArgs e)
+            }
+            currentBet -= allInBet;
+
+            //modify list of players that are eligible for the new main pot by removing player(s) that are all-in
+            List<int> allInPlayers = playerBets
+                .Select((value, index) => new { value, index })  // Keep both the value and the index
+                .Where(x => x.value == 0)  // Filter where the bet is 0 (all-in)
+                .Select(x => Pots[0].EligiblePlayers[x.index])  // Select the corresponding player from EligiblePlayers at the same index
+                .ToList();
+
+            numBets -= allInPlayers.Count;
+            playerBets.RemoveAll(value => value == 0);
+
+            //New side pot created. Pot that all-in player can win is shifted to inactive position, no more bets can be added to this pot
+            Pots.Add(new TexasHoldEmManager.TexasHoldEmPot(PotType.SIDE));
+            Pots[Pots.Count - 1].IncrementPot(Pots[0].Total); //shifting the pot that the all in player can win to the side, this pot will no longer be added to
+            Pots[0].DecrementPot(Pots[Pots.Count - 1].Total); //reseting active pot value to empty
+            Pots[Pots.Count - 1].EligiblePlayers = new List<int>(Pots[0].EligiblePlayers); //copys the list of players eligible to win the now side pot
+
+            //removing all players that are all in from eligibility from main pot and any pots created in the future
+            for (int player = 0; player < allInPlayers.Count; player++)
             {
-                return Pots.Select(pot => pot.Total).ToList();
+                Pots[0].RemoveEligiblePlayer(allInPlayers[player]);
             }
 
-            /// <summary>
-            /// Resets the pots list to an empty state.
-            /// </summary>
-            public void ResetPots()
+            //if there is a second player that has gone all in this round, recursively add pots until player that has called max bet has been reached
+            if (currentBet != playerBets.Min())
             {
-                Pots.Clear();
+                CreateSidePots(currentBet, playerBets);
+            }
+
+            //if all players but one are all-in 
+            else if (numBets == 1)
+            {
+                Pots.RemoveAt(0); //remove redundant pot creation
+                Pots[0].PotType = PotType.MAIN;
+            }
+
+            else
+            {
+                //add remainder of bets from players not all-in into new main pot
+                AddToPot(currentBet, playerBets);
             }
         }
-        #endregion Methods
+
+        /// <summary>
+        /// Pays out the winnings to the player for the individual pot. 
+        /// </summary>
+        /// <param name="winners">Number of players that have won the pot.</param>
+        public int DistributePot(int winners, int potNum)
+        {
+            if (winners == 0)
+            {
+                throw new ArgumentException("Number of winners cannot be zero.");
```

Comment:
under no circumstance should we throw an exception like this. instead, on the call to DistributePot, just confirm winners > 0 before calling and if not, just skip it because its errored out.

Throwing exceptions like this is scary, and can break the game if its something simple.